samples/peripheral/radio_test/src/radio_test.c

```diff
@@ -66,6 +68,50 @@ static uint8_t ppi_radio_start;
 static struct radio_test_fem fem;
 #endif /* CONFIG_FEM */
 
+static uint16_t channel_to_frequency(nrf_radio_mode_t mode, uint8_t channel)
+{
+#if CONFIG_HAS_HW_NRF_RADIO_IEEE802154
+	if (mode == NRF_RADIO_MODE_IEEE802154_250KBIT) {
+		if ((channel >= IEEE_MIN_CHANNEL) &&
+			(channel <= IEEE_MAX_CHANNEL)) {
+			return CHAN_TO_FREQ(IEEE_FREQ_CALC(channel));
+		} else {
+			return CHAN_TO_FREQ(IEEE_DEFAULT_FREQ);
+		}
+	} else {
+		return CHAN_TO_FREQ(channel);
+	}
+#else
+	return CHAN_TO_FREQ(channel);
+#endif /* CONFIG_HAS_HW_NRF_RADIO_IEEE802154 */
+}
+
+static void radio_power_set(nrf_radio_mode_t mode, uint8_t channel, int8_t power)
```

Comment:
Is it possible to get the +3 dBm mode for nRF5340 in here?
Similar to what is done in 802.15.4 implementation here:
https://github.com/nrfconnect/sdk-nrfxlib/blob/v2.1.2/nrf_802154/driver/src/nrf_802154_trx.c#L231-L245

Reply:
Yes it is even needed to ensure correct MPSL splitter functionality on the nRF53. Added

Reply:
Perfect, thank you! this worked like a charm, verified this earlier today in the lab with a spectrum analyzer.